js/ad_manager_vast.js

```diff
@@ -150,14 +351,53 @@ OO.Ads.manager(function(_, $) {
      * for different versions.
      * @private
      * @method Vast#supportsSkipAd
+     * @param {string} version The Vast version as parsed from the XML
      * @returns {boolean} true if the skip ad functionality is supported in the specified Vast version,
-     *                    false otherwise
+     *                    false otherwise.
      */
     var supportsSkipAd = _.bind(function(version) {
       return _.contains(SUPPORTED_FEATURES[getMajorVersion(version)], FEATURES.SKIP_AD);
     }, this);
 
     /**
+     * Helper function to grab error information. VastAdSingleParser already grabs error data while
+     * creating ad object, but some errors may occur before the object is created.
+     * Note: <Error> can only live in three places: directly under <VAST>, <Ad>, or <Wrapper> elements.
+     * <Error> tags are also optional so they may not always exist.
+     * @public
+     * @method Vast#getErrorInfo
+     * @param {xml} vastXML Contains the vast ad data to be parsed
+     * @param {string} wrapperParentIdArg Is the current ad's "parent" wrapper ID. Could be
+     * undefined if ad does not have parent/wrapper.
+     * @returns {object} The error object with a list of error urls and whether or not there are no ads.
+     */
+    this.getErrorInfo = _.bind(function(vastXML) {
```

Comment:
is this looking for errors of getting the error urls?  I'm a little confused whether this is doing validation or parsing.  Looks like mainly just parsing data.  Possibly consider renaming to ErrorTrackingInfo?